tools/content-generator/src/main/java/org/projectnessie/tools/contentgenerator/cli/RefreshContent.java

```diff
@@ -104,6 +117,37 @@ public void execute() throws BaseNessieClientServerException {
     }
   }
 
+  private void refreshAll(NessieApiV2 api) throws NessieNotFoundException {
+    try (Stream<Reference> refs = api.getAllReferences().stream()) {
+      refs.forEach(r -> refreshAll(api, r));
+    }
+  }
+
+  private void refreshAll(NessieApiV2 api, Reference ref) {
+    if (ref.getType() != Reference.ReferenceType.BRANCH) {
+      return;
+    }
+
+    try (Stream<EntriesResponse.Entry> entries =
+        api.getEntries().reference(ref).withContent(false).stream()) {
+      List<ContentKey> batch = new ArrayList<>(batchSize);
```

Comment:
Maybe replace all lines in the `try` with
`Iterators.partition(entries.iterator(), batchSize).forEachRemaining(batch -> refresh(api, ref, batch));`

Reply:
good point